### DISCUSSION TITLE
Self hosted runner "not a git repository"

### DISCUSSION THREAD

**Discussion by csumpter** [community/community#26172] [Actions]
<p>Error: “fatal: not a git repository (or any of the parent directories): .git”</p>
<p>I upgraded one of our self hosted runners from 2.162.0 to 2.165.2, and now the runner seems to be missing it’s git repository db when the action triggers and executes. No other changes have been made to the actual workflow files. I did ssh into the instance, and ensure that is is indeed actually missing the .git dir.</p>
<p>I am using the following checkout options:</p>
<pre><code>- uses: actions/checkout@v2
      with:
        fetch-depth: 1
</code></pre>
<p>Any ideas of what might be wrong here?</p>

**Comment by Yanjingzhu** [comment]
<p>Could you please enable step debug log and share the detail logs of checkout step?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/884a4b3a94e955a126e2153afa9bb2f98589e50b.png" data-download-href="/uploads/short-url/jrG6OfM1gVusJUuygJoozEuW9FF.png?dl=1" title="checkout log.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095922-b7b9e0d8-de7f-40e7-b2b5-614521c62941.png" alt="checkout log.png" data-base62-sha1="jrG6OfM1gVusJUuygJoozEuW9FF" width="690" height="454" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/884a4b3a94e955a126e2153afa9bb2f98589e50b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">checkout log.png</span><span class="informations">852×561 29.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And please go to the self-hosted runner work folder, please try to delete the repository name folder, in my case, my repo name is “actions-test”, so I delete folder actions-test. Will this delete operation work for you? </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181095923-6ab6776b-0c60-437a-b9da-4b577af8ce12.png" alt="runner work folder.png" data-base62-sha1="fTNjwbNkt3gmJC28Z2ayL57r2xZ" width="666" height="228"></p>

**Comment by csumpter** [comment]
<p>Thank you for the response!</p>
<p>This is on a new instance built off an AMI on AWS that had no pre-existing runner state. The <code>_work</code> dir was empty to begin with, but I did clear it out regardless. That does not fix the issue.</p>
<p>What docs exactly would you recommend I follow to enable those debug details? I looked around and saw varying references, but I dont think I am getting the same output that you are providing an example of.</p>
<p>I did downgrade the checkout action from v2 to v1, and the build works correctly, however it would be beneficial to figure out what the issue with the v2 checkout action is.</p>

**Comment by csumpter** [comment, marked as answer]
<p>The issue was that the resulting AMI had a old version of git installed: 2.17.1.</p>
<p>Solution: Upgrade the version of git installed on the self hosted runner.</p>
